src/overlays/actors/ovl_En_Jso/z_en_jso.c

```diff
@@ -81,106 +158,1547 @@ ActorInit En_Jso_InitVars = {
+    leftSwordBlureInit.p2StartColor[3] = rightSwordBlureInit.p2StartColor[3] = 64;
+    rightSwordBlureInit.p1EndColor[3] = leftSwordBlureInit.p1EndColor[3] = rightSwordBlureInit.p2EndColor[3] =
+        leftSwordBlureInit.p2EndColor[3] = 0;
+    rightSwordBlureInit.elemDuration = leftSwordBlureInit.elemDuration = 8;
+    rightSwordBlureInit.unkFlag = leftSwordBlureInit.unkFlag = 0;
+    rightSwordBlureInit.calcMode = leftSwordBlureInit.calcMode = 2;
+
+    Effect_Add(play, &this->rightSwordBlureIndex, EFFECT_BLURE1, 0, 0, &rightSwordBlureInit);
+    Effect_Add(play, &this->leftSwordBlureIndex, EFFECT_BLURE1, 0, 0, &leftSwordBlureInit);
+
+    this->actor.gravity = -3.0f;
+    this->scale = 0.035f;
+    this->actor.flags |= ACTOR_FLAG_CANT_LOCK_ON;
+    this->hintType = EN_JSO_GET_HINT_TYPE(&this->actor);
+    this->introCsType = this->hintType & EN_JSO_INTRO_SCALE_UP;
+    EnJso_SetupHandleIntroCutscene(this);
+}
+
+void EnJso_Destroy(Actor* thisx, PlayState* play) {
+    EnJso* this = THIS;
+    EnEncount3* parent;
+
+    Collider_DestroyCylinder(play, &this->bodyCollider);
+    Collider_DestroyQuad(play, &this->rightSwordCollider);
+    Collider_DestroyQuad(play, &this->leftSwordCollider);
+
+    if ((this->actor.parent != NULL) && (this->actor.parent->update != NULL)) {
+        parent = (EnEncount3*)this->actor.parent;
+        if (parent->actor.update != NULL) {
+            parent->child = NULL;
+            if (parent->unk14E > 0) {
+                parent->unk14E--;
+            }
+        }
+    }
+
+    Effect_Destroy(play, this->rightSwordBlureIndex);
+    Effect_Destroy(play, this->leftSwordBlureIndex);
+}
+
+void EnJso_ChangeAnim(EnJso* this, s32 animIndex) {
+    f32 morphFrames = -4.0f;
+
+    this->animEndFrame = Animation_GetLastFrame(sAnimations[animIndex]);
+    if ((animIndex == EN_JSO_ANIM_GUARD) || (animIndex == EN_JSO_ANIM_DAMAGED)) {
+        morphFrames = 0.0f;
+    }
+
+    Animation_Change(&this->skelAnime, sAnimations[animIndex], 1.0f, 0.0f, this->animEndFrame,
+                     sAnimationModes[animIndex], morphFrames);
+}
+
+void EnJso_SetupHandleIntroCutscene(EnJso* this) {
+    EnEncount3* parent = (EnEncount3*)this->actor.parent;
+
+    this->csId = parent->csId;
+    this->swordState = EN_JSO_SWORD_STATE_NONE_DRAWN;
+    this->action = EN_JSO_ACTION_HANDLE_INTRO_CUTSCENE;
+    this->actor.flags |= ACTOR_FLAG_100000;
+    this->actionFunc = EnJso_HandleIntroCutscene;
+}
+
+/**
+ * Responsible for handling all aspects of the intro cutscene, including manipulating the sub-camera, making the Garo
+ * appear in one of four different ways, activating the ring of fire, etc. When the cutscene is over, the Garo will
+ * jump back to start the fight.
+ */
+void EnJso_HandleIntroCutscene(EnJso* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    f32 curFrame = this->skelAnime.curFrame;
+    s16 showTextbox;
+    s16 diffToTargetRobeRightRot;
+
+    switch (this->introCsState) {
```

Comment:
Could you add a default to this case and maintain matching? (sometimes these

Reply:
No, I tried to insert a default case in every possible place in this switch-block, and it always breaks matching. It's the only switch in this file that refuses to be okay with a default case.